api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -96,6 +97,9 @@ public Obs saveObs(Obs obs, String changeMessage) throws APIException {
 		if(obs.getId() != null && changeMessage == null){
 			throw new APIException("Obs.error.ChangeMessage.required", (Object[]) null);
 		}
+		if(obs.getPerson()!=obs.getEncounter().getPatient().getPerson()) {
```

Comment:
Using != will simply compare the hashcode of the two objects, but it will not compare the values. In order to get desired result, you will have to use .equals() method